contracts/StateReceiver.sol

```diff
@@ -1,64 +1,40 @@
 pragma solidity ^0.5.11;
 
-import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
 import { RLPReader } from "solidity-rlp/contracts/RLPReader.sol";
 
 import { System } from "./System.sol";
-import { ValidatorVerifier } from "./ValidatorVerifier.sol";
 import { IStateReceiver } from "./IStateReceiver.sol";
-import { IterableMapping } from "./IterableMapping.sol";
 
-
-contract StateReceiver is System, ValidatorVerifier {
-  using SafeMath for uint256;
+contract StateReceiver is System {
   using RLPReader for bytes;
   using RLPReader for RLPReader.RLPItem;
 
-  // proposed states
-  IterableMapping.Map private proposedStates;
+  uint256 public lastStateSyncTime;
 
-  // states and proposed states
+  // stateId to isSynced
   mapping(uint256 => bool) public states;
 
-  // add pending state
-  function proposeState(
-    uint256 stateId
-  ) external {
-    // check if sender is validator
-   require(isValidator(msg.sender),"Invalid validator");
-
-    // check if state is already proposed
-    require(IterableMapping.contains(proposedStates, stateId) == false, "State already proposed");
-
-    // state should not prense
-    require(states[stateId] == false, "State already committed");
-
-    // propose state by adding it into proposed states
-    IterableMapping.insert(proposedStates, stateId, true);
-  }
-
   // commit new state
   function commitState(
-    bytes calldata recordBytes
+    bytes calldata recordBytes,
+    uint256 syncTime
```

Comment:
Suggestion: We should switch the position.

Reply:
Done.